process/caller.py

```diff
@@ -158,17 +190,32 @@ def call_models_and_write_output(self, xc: np.ndarray, ifail: int) -> None:
                     finalise(self.models, ifail)
                     return
 
-                # Mfiles not yet idempotent: re-evaluate models
+                # Mfiles not yet idempotent: need to re-evaluate models
                 logger.debug("Mfiles not idempotent, evaluating models again")
-                previous_mfile_arr = np.copy(current_mfile_arr)
+                previous_mfile_data = mfile_data.copy()
 
-            raise RuntimeError(
+            # Values haven't all stabilised after 10 evaluations
+            # Which variables are still changing?
+            warnings.warn(
                 "Model evaluations at the current optimisation parameter vector "
                 "don't produce idempotent values in the final output."
             )
+
+            print(
+                tabulate(
+                    [[k, v[0], v[1]] for k, v in nonconverged_vars.items()],
+                    headers=["Variable", "Previous value", "Current value"],
+                )
+            )
```

Comment:
Should we consider adding some colour here? It can be quite hard to see these warnings and tables in all of the other PROCESS junk (as seen below)

```
...
*Cryostat roof allowance includes uppermost PF coil and outer thermal shield.
*Cryostat floor allowance includes lowermost PF coil, outer thermal shield and gravity support.
This is longer than 110 columns.
 Warning in routine OCMMNT :

*Cryostat roof allowance includes uppermost PF coil and outer thermal shield.
*Cryostat floor allowance includes lowermost PF coil, outer thermal shield and gravity support.
This is longer than 110 columns.
 Warning in routine OCMMNT :

*Cryostat roof allowance includes uppermost PF coil and outer thermal shield.
*Cryostat floor allowance includes lowermost PF coil, outer thermal shield and gravity support.
This is longer than 110 columns.
/home/ubuntu/PROCESS/process/caller.py:197: UserWarning: Model evaluations at the current optimisation parameter vector don't produce idempotent values in the final output.
  warnings.warn(
Variable      Previous value    Current value
----------  ----------------  ---------------
coe                      nan              nan
 Warning in routine OCMMNT :

*Cryostat roof allowance includes uppermost PF coil and outer thermal shield.
*Cryostat floor allowance includes lowermost PF coil, outer thermal shield and gravity support.
This is longer than 110 columns.
...
```

Reply:
Yellow warning colour added.